tests/asconfig/complicated/assembly/index.ts

```diff
@@ -0,0 +1,178 @@
+      v2 += v3;
+      v1 = rotl<u64>(v1, 13);
+      v3 = rotl<u64>(v3, 16);
+      v1 ^= v0;
+      v3 ^= v2;
+      v0 = rotl<u64>(v0, 32);
+      v2 += v1;
+      v0 += v3;
+      v1 = rotl<u64>(v1, 17);
+      v3 = rotl<u64>(v3, 21);
+      v1 ^= v2;
+      v3 ^= v0;
+      v2 = rotl<u64>(v2, 32);
+
+      // compression algorithm
+      v0 += v1;
+      v2 += v3;
+      v1 = rotl<u64>(v1, 13);
+      v3 = rotl<u64>(v3, 16);
+      v1 ^= v0;
+      v3 ^= v2;
+      v0 = rotl<u64>(v0, 32);
+      v2 += v1;
+      v0 += v3;
+      v1 = rotl<u64>(v1, 17);
+      v3 = rotl<u64>(v3, 21);
+      v1 ^= v2;
+      v3 ^= v0;
+      v2 = rotl<u64>(v2, 32);
+
+      v0 ^= mi;
+
+      mp += 8;
+    }
+
+    store<u8>(buf, ml, 7);
+
+    let ic: usize = 0;
+    while (<usize>mp < ml) {
+      store<u8>(buf + (ic++), load<u8>(m + (mp++)));
+    }
+
+    while (ic < 7) {
+      store<u8>(buf + (ic++), 0);
+    }
+    let mil = load<u64>(buf);
+
+    v3 ^= mil;
+
+    // compression algorithm
+    v0 += v1;
+    v2 += v3;
+    v1 = rotl<u64>(v1, 13);
+    v3 = rotl<u64>(v3, 16);
+    v1 ^= v0;
+    v3 ^= v2;
+    v0 = rotl<u64>(v0, 32);
+    v2 += v1;
+    v0 += v3;
+    v1 = rotl<u64>(v1, 17);
+    v3 = rotl<u64>(v3, 21);
+    v1 ^= v2;
+    v3 ^= v0;
+    v2 = rotl<u64>(v2, 32);
+
+    // compression algorithm
+    v0 += v1;
+    v2 += v3;
+    v1 = rotl<u64>(v1, 13);
+    v3 = rotl<u64>(v3, 16);
+    v1 ^= v0;
+    v3 ^= v2;
+    v0 = rotl<u64>(v0, 32);
+    v2 += v1;
+    v0 += v3;
+    v1 = rotl<u64>(v1, 17);
+    v3 = rotl<u64>(v3, 21);
+    v1 ^= v2;
+    v3 ^= v0;
+    v2 = rotl<u64>(v2, 32);
+
+    v0 ^= mil;
+    v2 ^= 0xff;
+
+    // compression algorithm
+    v0 += v1;
+    v2 += v3;
+    v1 = rotl<u64>(v1, 13);
+    v3 = rotl<u64>(v3, 16);
+    v1 ^= v0;
+    v3 ^= v2;
+    v0 = rotl<u64>(v0, 32);
+    v2 += v1;
+    v0 += v3;
+    v1 = rotl<u64>(v1, 17);
+    v3 = rotl<u64>(v3, 21);
+    v1 ^= v2;
+    v3 ^= v0;
+    v2 = rotl<u64>(v2, 32);
+
+    // compression algorithm
+    v0 += v1;
+    v2 += v3;
+    v1 = rotl<u64>(v1, 13);
+    v3 = rotl<u64>(v3, 16);
+    v1 ^= v0;
+    v3 ^= v2;
+    v0 = rotl<u64>(v0, 32);
+    v2 += v1;
+    v0 += v3;
+    v1 = rotl<u64>(v1, 17);
+    v3 = rotl<u64>(v3, 21);
+    v1 ^= v2;
+    v3 ^= v0;
+    v2 = rotl<u64>(v2, 32);
+
+    // compression algorithm
+    v0 += v1;
+    v2 += v3;
+    v1 = rotl<u64>(v1, 13);
+    v3 = rotl<u64>(v3, 16);
+    v1 ^= v0;
+    v3 ^= v2;
+    v0 = rotl<u64>(v0, 32);
+    v2 += v1;
+    v0 += v3;
+    v1 = rotl<u64>(v1, 17);
+    v3 = rotl<u64>(v3, 21);
+    v1 ^= v2;
+    v3 ^= v0;
+    v2 = rotl<u64>(v2, 32);
+
+    // compression algorithm
+    v0 += v1;
+    v2 += v3;
+    v1 = rotl<u64>(v1, 13);
+    v3 = rotl<u64>(v3, 16);
+    v1 ^= v0;
+    v3 ^= v2;
+    v0 = rotl<u64>(v0, 32);
+    v2 += v1;
+    v0 += v3;
+    v1 = rotl<u64>(v1, 17);
+    v3 = rotl<u64>(v3, 21);
+    v1 ^= v2;
+    v3 ^= v0;
+    v2 = rotl<u64>(v2, 32);
+
+    return v0 ^ v1 ^ v2 ^ v3;
+  }
+}
```

Comment:
Can this file be shortened somehow, to just contain the parts actually needed for the test?

Reply:
The funny thing is that I had committed this before you suggested to skip it 😅 . Perhaps we could just remove it as you suggest.

Reply:
Better. We can check to make sure that initial memory is checked using `memory.size()`. That seemed to be the most complicated thing I can think of. Sure enough, I found a merge order bug.

In any case, this is all ready for review again.